sympy/solvers/solveset.py

```diff
@@ -115,7 +115,17 @@ def invert_real(f_x, y, x, domain=S.Reals):
     Inverts a real-valued function. Same as _invert, but sets
     the domain to ``S.Reals`` before inverting.
     """
-    return _invert(f_x, y, x, domain)
+    lhs, rhs_s = _invert(f_x, y, x, domain)
+    if lhs == x and isinstance(rhs_s, FiniteSet):
+        for rhs in rhs_s:
+            try:
+                if checksol(f_x - y, x, rhs):
+                    pass
+                else:
+                    return (f_x, FiniteSet(y))
```

Comment:
One of the design features that I think is good about solveset is that it doesn't have to use solution checking--as I recall. It seems to me that a simple check in the function that tried to return 1/log(0) would be a better fix: if the base is 0 don't return that solution (or do something else). (And if it can't be determined if the base is 0 or not, return a conditionset solution.)